mailet/standard/src/test/java/org/apache/james/transport/mailets/MailAttributesListToMimeHeadersTest.java

```diff
@@ -0,0 +1,132 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class MailAttributesListToMimeHeadersTest {
+
+    @Rule public ExpectedException expectedException = ExpectedException.none();
+
+    private Mailet mailet;
+
+    private final String HEADER_NAME1 = "JUNIT";
+    private final String HEADER_NAME2 = "JUNIT2";
+
+    private final ImmutableList<String> MAIL_ATTRIBUTE_VALUE1 = ImmutableList.of("test1.1", "test1.2");
+    private final ImmutableList<String> MAIL_ATTRIBUTE_VALUE2 = ImmutableList.of("test2.1", "test2.2");
+
+    private final String MAIL_ATTRIBUTE_NAME1 = "org.apache.james.test";
+    private final String MAIL_ATTRIBUTE_NAME2 = "org.apache.james.test2";
+
+    @Before
+    public void setup() {
+        mailet = new MailAttributesListToMimeHeaders();
+    }
+
+    @Test
+    public void shouldPutAttributesIntoHeadersWhenMappingDefined() throws MessagingException {
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("simplemapping",
+                        MAIL_ATTRIBUTE_NAME1 + "; " + HEADER_NAME1 +
+                        "," + MAIL_ATTRIBUTE_NAME2 + "; " + HEADER_NAME2 +
+                        "," + "another.attribute" + "; " + "Another-Header")
+                .build();
+        mailet.init(mailetConfig);
+
+        FakeMail mockedMail = MailUtil.createMockMail2Recipients(MailUtil.createMimeMessage());
+        mockedMail.setAttribute(MAIL_ATTRIBUTE_NAME1, MAIL_ATTRIBUTE_VALUE1);
+        mockedMail.setAttribute(MAIL_ATTRIBUTE_NAME2, MAIL_ATTRIBUTE_VALUE2);
+        mockedMail.setAttribute("unmatched.attribute", "value");
+
+        mailet.service(mockedMail);
+
+        assertThat(mockedMail.getMessage().getHeader(HEADER_NAME1)).containsAll(MAIL_ATTRIBUTE_VALUE1);
+        assertThat(mockedMail.getMessage().getHeader(HEADER_NAME1)).hasSameSizeAs(MAIL_ATTRIBUTE_VALUE1);
+        assertThat(mockedMail.getMessage().getHeader(HEADER_NAME2)).containsAll(MAIL_ATTRIBUTE_VALUE2);
+        assertThat(mockedMail.getMessage().getHeader(HEADER_NAME2)).hasSameSizeAs(MAIL_ATTRIBUTE_VALUE2);
+    }
+
+    @Test
+    public void shouldAddAttributeIntoHeadersWhenHeaderAlreadyPresent() throws MessagingException {
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("simplemapping", MAIL_ATTRIBUTE_NAME1 + "; " + HEADER_NAME1)
+                .build();
+        mailet.init(mailetConfig);
+
+        FakeMail mockedMail = MailUtil.createMockMail2Recipients(MimeMessageBuilder.mimeMessageBuilder()
+            .addHeader(HEADER_NAME1, "first value")
+            .build());
+        mockedMail.setAttribute(MAIL_ATTRIBUTE_NAME1, MAIL_ATTRIBUTE_VALUE1);
+
+        mailet.service(mockedMail);
+
+        assertThat(mockedMail.getMessage().getHeader(HEADER_NAME1)).containsAll(MAIL_ATTRIBUTE_VALUE1);
```

Comment:
You should be asserting it contains only the Union(MAIL_ATTRIBUTE_VALUE1, "first value") in a single assert.